src/app/wallet/wallet/send/send.service.ts

```diff
@@ -26,42 +27,64 @@ export class SendService {
   }
 
   /* Sends a transaction */
-  public sendTransaction(
-    input: string, output: string, address: string,
-    amount: number, comment: string, narration: string,
-    ringsize: number, numsignatures: number, substractfeefromamount: boolean) {
+  // @TODO: estimate fee should be change
+  public sendTransaction(tx: TransactionBuilder) {
+    tx.comment = tx.narration = tx.note;
 
-    const rpcCall: string = this.getSendRPCCall(input, output);
+    const rpcCall: string = this.getSendRPCCall(tx.input, tx.output);
     const anon: boolean = this.isAnon(rpcCall);
     const params: Array<any> = this.getSendParams(
-      anon, address, amount, comment,
-      narration, ringsize, numsignatures, substractfeefromamount);
+      anon, tx.toAddress, tx.amount, tx.comment,
+      tx.narration, tx.ringsize, tx.numsignatures, tx.subtractFeeFromAmount);
 
     this._rpc.call('send' + rpcCall, params)
       .subscribe(
-      success => this.rpc_send_success(success, address, amount),
-      error => this.rpc_send_failed(error.message, address, amount));
+        success => this.rpc_send_success(success, tx.toAddress, tx.amount),
+        error => this.rpc_send_failed(error.message, tx.toAddress, tx.amount));
   }
 
-  public transferBalance(
-    input: string, output: string,
-    amount: number, ringsize: number, numsignatures: number, substractfeefromamount: boolean) {
+  public getTransactionFee(tx: TransactionBuilder): Observable<any> {
+    if (!tx.toAddress) {
+      return new Observable((observer) => {
+        this.getDefaultStealthAddress().take(1).subscribe(
+          (stealthAddress: string) => {
+            this._rpc.call('sendtypeto', ['part', 'part', [{
+              subfee: true,
+              address: stealthAddress,
+              amount: tx.amount,
+            }], '', '', 8, 64, true]).subscribe(fee => {
+              observer.next(fee);
+              observer.complete();
+            });
+          });
+      });
+    } else {
+      return this._rpc.call('sendtypeto', ['part', 'part', [{
+        subfee: true,
+        address: tx.toAddress,
+        amount: tx.amount,
+      }], '', '', 8, 64, true]).map(
```

Comment:
I have to check again what this 8 and 64 mean exactly, I recall these being the default values.